FrontEnd/src/BookingPage/BookingDB.js

```diff
@@ -8,85 +8,123 @@ import {
+    const appointmentDataWithStatus = {
       ...appointmentData,
+      customerUID: userUID,
       vendorUID,
+      vendorUsername,
       customerUsername,
       Status,
     };
 
     const userDocRef = doc(database, DATABASE_FOLDER_NAME, userUID);
-    const userAppointmentCollection = collection(userDocRef, "Appointments");
-    const userAppointmentDocRef = doc(
-      userAppointmentCollection,
-      userAppointmentDataWithStatus.appointmentTitle
+    const userAppointmentCollection = collection(
+      userDocRef,
+      APPOINTMENT_COLLECTION
     );
-    await setDoc(userAppointmentDocRef, userAppointmentDataWithStatus);
+    await addDoc(userAppointmentCollection, appointmentDataWithStatus);
 
     const vendorDocRef = doc(database, DATABASE_FOLDER_NAME, vendorUID);
     const vendorAppointmentCollection = collection(
       vendorDocRef,
-      "Appointments"
-    );
-    const vendorAppointmentDocRef = doc(
-      vendorAppointmentCollection,
-      vendorAppointmentDataWithStatus.appointmentTitle
+      APPOINTMENT_COLLECTION
     );
-    await setDoc(vendorAppointmentDocRef, vendorAppointmentDataWithStatus);
+    await addDoc(vendorAppointmentCollection, appointmentDataWithStatus);
   } catch (error) {
     throw new Error(`Error booking an appointment: ${error.message}`);
   }
 };
 
-const vendorAppointmentNotification = (vendorId, callback) => {
+const appointmentChanges = (userID) => {
+  const userAppointmentCollectionRef = collection(
+    database,
+    DATABASE_FOLDER_NAME,
+    userID,
+    APPOINTMENT_COLLECTION
+  );
+
+  const unsubscribe = onSnapshot(userAppointmentCollectionRef, (snapshot) => {
+    snapshot.docChanges().forEach((change) => {
+      const appointmentData = change.doc.data();
+      const vendorID = appointmentData.vendorUID;
+
+      if (change.type === "added") {
+        const notificationMessage = `You just requested an appointment on ${appointmentData.appointmentDate} with ${appointmentData.vendorUsername} for ${appointmentData.appointmentTitle}`;
+        addNotification(userID, notificationMessage, appointmentData);
+
+        const vendorNotificationMessage = `A new service request has been made by ${appointmentData.customerUsername} for ${appointmentData.appointmentTitle}`;
+        addNotification(vendorID, vendorNotificationMessage, appointmentData);
+      } else if (
+        change.type === "modified" &&
+        appointmentData.Status === "accepted"
+      ) {
+        const notificationMessage = `Your request has been accepted by ${appointmentData.vendorUsername}`;
+        addNotification(userID, notificationMessage, appointmentData);
+
+        const vendorNotificationMessage = `You accepted a service request from ${appointmentData.customerUsername}`;
+        addNotification(vendorID, vendorNotificationMessage, appointmentData);
+      }
+    });
+  });
+
+  return unsubscribe;
+};
+
+const addNotification = async (ID, message, appointmentData) => {
```

Comment:
Should this be renamed from `addNotification` to `sendNotification`? Since this is sending a notification to the user.

Reply:
"addNotification" adds notification data to the Notification database. "fetchNotifications" is what sends the notification to users